File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/FieldRepository.java
package com.github.ol_loginov.heaplibweb.repository.heap;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.stream.Stream;

@Repository
public interface FieldRepository extends CrudRepository<FieldEntity, Integer> {
	Stream<FieldEntity> streamAllByHeapIdAndDeclaringClassId(int heapId, long declaringClassId);

	Stream<FieldEntity> streamAllByHeapId(int heapId);
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/FieldRepository.kt
package com.github.ol_loginov.heaplibweb.repository.heap

import java.util.stream.Stream

interface FieldRepository {
    fun persist(entity: FieldEntity)
    fun persistAll(entities: List<FieldEntity>)

    fun findAllByDeclaringClassIdOrderById(declaringClassId: Long): List<FieldEntity>
    fun streamAllOrderById(): Stream<FieldEntity>

    fun findById(id: Int): FieldEntity?
    fun findByDeclaringClassIdAndName(declaringClassId: Long, name: String): FieldEntity?
}